cmd/puppeth/ssh.go

```diff
@@ -77,7 +77,23 @@ func dial(server string, pubkey []byte) (*sshClient, error) {
 		log.Warn("No SSH key, falling back to passwords", "path", path, "err", err)
 	} else {
 		key, err := ssh.ParsePrivateKey(buf)
-		if err != nil {
+		if string(err.Error()) == "ssh: cannot decode encrypted private keys" {
+
+			fmt.Printf("Enter the passphrase to decrypt your ssh key (no echo):\n>")
+			blob, err := terminal.ReadPassword(int(syscall.Stdin))
+			fmt.Println()
+			if err != nil {
+				log.Warn("Couldn't read password to decrypt ssh key")
+			}
+
+			key, err := ssh.ParsePrivateKeyWithPassphrase(buf, []byte(blob))
+			if err != nil {
+				log.Warn("Bad SSH key, falling back to passwords", "path", path, "err", err)
```

Comment:
Perhaps instead of `Bad SSH key`, output `Failed to decrypt SSH key` here, since it better highlights the issue.